src/decorators/BIPStateDecorator/Core/BIPStateDecorator.Core.js

```diff
@@ -35,32 +35,39 @@ define([
 
             this.skinParts.$svg = $(template);
             this.skinParts.$name = this.skinParts.$svg.find('.name');
-
-            //TODO: If we need to show a pop-over (for showing the full name or some other data).
-            // this.skinParts.$svgContainer.popover({
-            //     delay: {
-            //         show: 150,
-            //         hide: 0
-            //     },
-            //     animation: false,
-            //     trigger: 'hover',
-            //     title: 'Multiplicity: ' + name.multiplicity,
-            //     content: 'Degree : ' + name.degree
-            // });
             this.skinParts.$svgContainer.append(this.skinParts.$svg);
         }
 
         if (this.name) {
             // TODO: Find the threshold and how to cut it when it exceeds.
-            if (this.name.length < 10) {
+            if (this.name.length < 8) {
                 this.skinParts.$name.text(this.name);
             } else {
-                this.skinParts.$name.text('...');
+                this.skinParts.$name.text(this.name.substring(0, 5).concat('...'));
+
+                //TODO: If we need to show a pop-over (for showing the full name or some other data).
+                this.skinParts.$svgContainer.popover({
```

Comment:
We should double-check that registering the popover more than once isn't an issue..